### DISCUSSION TITLE
JDK 20

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#263] [Q&A] [answered]
Hi guys.
We are using JDK 20 for the development of vector indexes.

Are there any obstacles in the integration of TornadoVM with JDK 20?
Will new byte code with versions higher than for JDK 17 be supported?
Are there any instability risks in production?

**Comment by andrii0lomakin** [comment]
And the follow-up question - is it safe to run TornadoVM with Loom virtual threads?

**Reply by jjfumero** [reply, marked as answer]
We haven't tested it yet, but from my view, this should work. TornadoVM can be invoked from a Java thread or from a Loom thread. Then the TornadoVM runtime will launch the application on the GPU using a low-level set of threads (CUDA thread, or OpenCL/SPIR-V threads). So TornadoVM is orthogonal to Java/Loom thread from the Java platform.

**Reply by andrii0lomakin** [reply]
Thank you. It will be interesting to experiment if VTs could be used to amortize the latency of interaction between device and host by batching user requests and executing them in one sweep.

**Comment by jjfumero** [comment]
Hi @laa , TornadoVM can't run with JDK 20 at the moment. However, we are actively working on this port (we had to update our JIT compiler to connect with the latest JVMCI and GraalVM 23.0). This is a work in progress and hopefully soon this will be fully supported. 

In the meantime, TornadoVM can be used with JDK 11-19. We use JDK 17 internally for development.

**Comment by andrii0lomakin** [comment]
Thank you, guys, for the prompt answer. Will keep tracking changes to use JDK 20 support once it is implemented.
